swag/client/ui/yfu_view.py

```diff
@@ -0,0 +1,272 @@
+import asyncio
+import disnake
+from swag.blocks.swag_blocks import Transaction
+from swag.id import CagnotteId, UserId
+from swag.yfu import Yfu
+from swag.blocks.yfu_blocks import RenameYfuBlock
+
+from .IHS_toolkit import *
+
+##TODO apparition du bouton "Renommer" et "Activer" dynamique
+class YfuNavigation(disnake.ui.View):
+    def __init__(self, swag_client, user_id, first_yfu_id):
+        super().__init__(timeout=None)
+
+        self.swag_client = swag_client
+        self.user_id = user_id
+        self.yfu_ids = sort_yfus_id(
+            swag_client.swagchain.account(self.user_id).yfu_wallet
+        )
+        self.yfus = [swag_client.swagchain.yfu(yfu_id) for yfu_id in self.yfu_ids]
+
+        # Generation des options du dropdown de waifu
+        for option in yfus_to_select_options(self.yfus):
+            self.dropdown_yfu.append_option(
+                option
+            )  ##TODO gérer quand il y a plus de 25 options
```

Comment:
May I ask you to put those kind of comments before or after the line, instead of at the end of a line?

Reply:
Resolved in 4372604 ("Moving TODO", 2022-02-21)